validphys2/src/validphys/process_options.py

```diff
@@ -217,6 +220,21 @@ def _dywboson_xq2map(kin_dict):
     return np.clip(x, a_min=None, a_max=1, out=x), np.concatenate((mass2, mass2))
 
 
+def _dyllZboson_xq2map(kin_dict):
+    """
+    Computes x and q2 mapping for rapidity observables
+    originating from a Z boson DY process.
+    """
+    mass2 = kin_dict[_Vars.m_Z2]
+    sqrts = kin_dict[_Vars.sqrts]
+    abs_yll = kin_dict[_Vars.abs_yll]
+
+    x1 = np.sqrt(mass2) / sqrts * np.exp(-abs_yll)
+    x2 = np.sqrt(mass2) / sqrts * np.exp(abs_yll)
+    x = np.concatenate((x1, x2))
+    return np.clip(x, a_min=None, a_max=1, out=x), np.concatenate((mass2, mass2))
+
```

Comment:
Since (as it should be) Z and W share this logic I think it would be better to put them into one single function.

Whether to abstract the logic out and have two calls of the same function with different variables or a single function that uses `get_one_of` it's up to you.
I think I'd favour the `get_one_of` because I have the feeling that there might be datasets (for `Z`) that use rapidity instead of absolute rapidity and maybe even pseudorapidity...